docs/troubleshooting.md

```diff
@@ -0,0 +1,115 @@
+ INFO  DATADOG: Datadog SDK is tracking errors
+ DEBUG  DATADOG: Starting RUM View “Products” #Products-oaZlP_FVwGM5vtPoup_rT
+ DEBUG  DATADOG: Adding RUM Action “RCTView” (TAP)
+```
+
+In this example, the first 4 logs indicate that the SDK has been correctly configured, the last 2 ones that some events are supposed to be sent.
+
+#### Possible cause of issue
+
+If on iOS you see some DEBUG logs indicating that logs or RUM events were sent **before** the initialization logs, it might be the cause for the SDK not sending events.
+
+Currently the SDK does not support sending events before initialization, and attempting to do so will put the SDK in a state where it cannot send any data.
+
+#### Solution
+
+##### With `DdSdkReactNative.initialize`
+
+If you use `DdSdkReactNative.initialize` to start the Datadog SDK, call this function in your top-level `index.js` file so the SDK is initialized before your other events are sent.
+
+##### With `DatadogProvider`
+
+Starting from SDK version `1.2.0-beta1`, you can initialize the SDK using the `DatadogProvider` component, which includes a RUM events buffer to prevent this very issue from happening.
+
+To use it, see our [migration guide](./migrating_to_datadog_provider.md)
+
+### Look at native logs
+
+Looking at native logs can give you more input on what could be going wrong.
+
+#### On iOS
+
+-   Open your project in xcode by running `xed ios`
+-   Build your project for a simulator or a device
+-   Native logs will appear on the bottom right corner:
+
+![](./xcode-logs.png)
+
+You can filter logs by “DATADOG” and look for any error.
+
+If you are indeed sending events, you should see the following logs:
+
+```
+[DATADOG SDK] 🐶 → 10:02:47.398 [DEBUG] ⏳ (rum) Uploading batch...
+[DATADOG SDK] 🐶 → 10:02:47.538 [DEBUG]    → (rum) accepted, won't be retransmitted: [response code: 202 (accepted), request ID: AAAABBBB-1111-2222-3333-777788883333]
+```
+
+The first one indicates that some data is being sent, the second one that the data has been received.
+
+#### Possible cause of issue
+
+If you see the log below, it means that you have called a RUM method before initialising the SDK.
+
+```
+[DATADOG SDK] 🐶 → 10:09:13.621 [WARN] The `Global.rum` was called but no `RUMMonitor` is registered. Configure and register the RUM Monitor globally before invoking the feature:
+```
+
+##### Solution with `DdSdkReactNative.initialize`
+
+If you use `DdSdkReactNative.initialize` to start the Datadog SDK, call this function in your top-level `index.js` file so the SDK is initialized before your other events are sent.
+
+###### Solution with `DatadogProvider`
+
+Starting from SDK version `1.2.0-beta1`, you can initialize the SDK using the `DatadogProvider` component, which includes a RUM events buffer to prevent this very issue from happening.
+
+To use it, see our [migration guide](./migrating_to_datadog_provider.md)
+
+### On Android
+
+-   Install [pidcat](https://github.com/JakeWharton/pidcat)
```

Comment:
An optional step. Normal logcat would work fine as well no?

Reply:
Yes indeed, is it ok if I mention it as a recommended step, as I believe this makes debugging much easier?